packages/framework/src/Model/Mail/Mailer.php

```diff
@@ -89,8 +96,11 @@ protected function getMessageWithReplacedVariables(MessageData $messageData)
         $message->setContentType('text/plain; charset=UTF-8');
         $message->setBody(htmlspecialchars_decode(strip_tags($body), ENT_QUOTES), 'text/plain');
         $message->addPart($body, 'text/html');
-        foreach ($messageData->attachmentsFilepaths as $attachmentFilepath) {
-            $message->attach(Swift_Attachment::fromPath($attachmentFilepath));
+
+        foreach ($messageData->attachments as $attachment) {
+            $swiftAttachment = Swift_Attachment::fromPath($this->mailTemplateFacade->getMailTemplateAttachmentFilepath($attachment));
+            $swiftAttachment->setFilename($attachment->getName() . '.' . $attachment->getExtension());
```

Comment:
e.g. here you are using `$attachment->getName() . '.' . $attachment->getExtension()` again. Another reason for putting it into method.

Reply:
Fixed